sklearn/cluster/bicluster.py

```diff
@@ -139,22 +139,23 @@ def _svd(self, array, n_components, n_discard):
                                       **kwargs)
 
         elif self.svd_method == 'arpack':
-            u, _, vt = svds(array, k=n_components, ncv=self.n_svd_vecs)
+            v0 = _init_arpack_v0(min(array.shape), self.random_state)
+            u, _, vt = svds(array, k=n_components, ncv=self.n_svd_vecs, v0=v0)
             if np.any(np.isnan(vt)):
                 # some eigenvalues of A * A.T are negative, causing
                 # sqrt() to be np.nan. This causes some vectors in vt
                 # to be np.nan.
                 A = safe_sparse_dot(array.T, array)
-                random_state = check_random_state(self.random_state)
-                # initialize with [-1,1] as in ARPACK
-                v0 = random_state.uniform(-1, 1, A.shape[0])
+                if A.shape[0] != min(array.shape):
```

Comment:
This if branch wasn't there before, why is it necessary now?

Reply:
It might  not be necessary. But since a new initialisation has been added a few line above (right before svds ), it’s to avoid initializing v0 twice in some cases (as in those case v0´s shape is ok. But it’s not pretty so i’m not convinced.

Reply:
Nah I was not convinced by this line in the first place. I think the impact of a double initialization is epsilon. I’ll just remove those lines and settle for a double init. Thanks for your input